js/actions/tests/loadSourceText.js

```diff
@@ -23,20 +29,111 @@ const mockThreadClient = {
+    return this.request;
+  }
+};
+
 describe("loadSourceText", () => {
-  beforeEach(function() {
-    this.store = createStore(mockThreadClient);
+  describe("loading one source text", function() {
+    beforeEach(function(done) {
+      this.store = createStore(simpleMockThreadClient);
+      this.store.dispatch(actions.loadSourceText({ actor: "foo1" }))
+        .then(() => done());
+    });
+
+    it("Store has the source text", function() {
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo1");
+      expect(fooSourceText.get("text")).to.equal(sourceText.foo1.source);
+    });
   });
 
-  describe("loading one source text", function() {
+  describe("loading two different sources", function() {
     beforeEach(function(done) {
+      this.store = createStore(simpleMockThreadClient);
       this.store.dispatch(actions.loadSourceText({ actor: "foo1" }))
+        .then(() => {
+          return this.store.dispatch(actions.loadSourceText({ actor: "foo2" }));
+        })
         .then(() => done());
     });
 
+    it("Store has the first source text", function() {
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo1");
+      expect(fooSourceText.get("text")).to.equal(sourceText.foo1.source);
+    });
+
+    it("Store has the second source text", function() {
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo2");
+      expect(fooSourceText.get("text")).to.equal(sourceText.foo2.source);
+    });
+  });
+
+  describe("loading a source twice", function() {
+    beforeEach(function(done) {
+      const store = createStore(simpleMockThreadClient);
+      this.store = store;
+
+      Task.spawn(function* () {
+        yield store.dispatch(actions.loadSourceText({ actor: "foo1" }));
+        yield store.dispatch(actions.loadSourceText({ actor: "foo1" }));
+        done();
+      });
+    });
+
     it("Store has the source text", function() {
-      const fooSourceText = queries.getSourceText(this.store.getState(), "foo1");
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo1");
       expect(fooSourceText.get("text")).to.equal(sourceText.foo1.source);
     });
   });
+
+  describe("source is loading", function() {
+    beforeEach(function(done) {
+      this.store = createStore(deferredMockThreadClient);
+      this.store.dispatch(actions.loadSourceText({ actor: "foo1" }));
+      // We're intentionally leaving the source promise pending
+      done();
+    });
+
+    it("Store has a loading source text", function() {
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo1");
+      expect(fooSourceText.get("loading")).to.equal(true);
+    });
+  });
+
+  describe("source failed to load", function() {
+    beforeEach(function(done) {
```

Comment:
I developed this style in part to avoid tests growing over time. Where the `beforeEach` handles the setup and the `it` handle the assertions. Often times, you might want to check a couple things and that way the test read, `when a source fails to load`, `the editor shows the error`, and `the error is saved in the store's source field`... The screenshot shows this a bit.

With that said, I'm okay putting the action in the `it` now as well.